### PR TITLE
Documentation Cleanup: Fix index.md link to decomposition.md

### PR DESCRIPTION
**Pull request by JKomoroski**
Documentation Cleanup: decomposition.md

Been a few months since I last worked on the documentation.

### CODE REVIEW

File: src/doc/docs/decomposition.md [hunk 1]
```diff
@@ -3,60 +3,58 @@
...
+modules (e.g. the [jdbi](extras/jdbi3.md) wrapper around the [dropwizard module](https://www.dropwizard.io/en/latest/manual/jdbi3.html)). 
+Such wrappers usually provide additional features and enhancements that would be impossible with a single module type.
 
 ## Guicey bundle
 
-As it [was described](concepts.md#bundles), guicey have to introduce its own [bundle](guide/bundles.md) because guicey provides
-additional configuration features (even in the simplest case you should be able to
-configure guice modules).
+As it [was described](concepts.md#bundles), guicey introduces its own [bundle](guide/bundles.md) because guicey provides
+additional configuration features. Even the simple use cases require some configuration of guice modules.
```
Comment (xvik) [inline src/doc/docs/decomposition.md:29]: "Even the simple use cases require some configuration of guice modules." -> No, it's not strictly required, but may be requirwed - the simplest example is a need of gucie module registration from bundle. Maybe it should be like: "In the simplest case, you may need to apply guice modules from bundle"

File: src/doc/docs/decomposition.md [hunk 2]
```diff
@@ -181,41 +179,42 @@ public class ModuleImpl<C extends Configuration> extends DropwizardAwareModule<C
...
+may be used. This could be used to de-duplicate only instances with the same constructor arguments.
 
-It may be mostly important for [guice modules](guide/deduplication.md#guice-modules) as guice will not start with duplicate bundings
+This is most applicable to [guice modules](guide/deduplication.md#guice-modules) as guice will not start with duplicate bindings
 (`MyModule extends UniqueModule`).
 
 ### Auto-loaded bundle
 
-Auto loading based on guicey [bundles lookup](guide/bundles.md#service-loader-lookup).
+Auto-loading is based on guicey [bundles lookup](guide/bundles.md#service-loader-lookup) algorithm.
```
Comment (xvik) [inline src/doc/docs/decomposition.md:197]: I may not know some language nuances, but when I see an "algorithm" I imagine something sophisticated. Bundles lookup implementation is very simple (lookup and load) I don't think it deserves to be called an "algorithm". Probably a "feature" would be better.

File: src/doc/docs/decomposition.md [hunk 3]
```diff
@@ -244,9 +243,8 @@ public class MyFeatureBundle implements GuiceyBundle {
 As bundle has almost complete access to configuration, it can use [disables](guide/disables.md)
 to substitute application functions.
 
-For example, it is known that application use `ServiceX` (from some core module in organization),
-but this bundle requires modified service. It can disable core module, installing feature 
-and register customized module:
+For example, it is known that an application uses `ServiceX` from some core module provided by the organization. Your module 
+requires a modified service. Your bundle may disable the core module, and install a replacement customized module:
```
Comment (xvik) [inline src/doc/docs/decomposition.md:247]: "and install a replacement customized module" -> I think its not clear what is replacement and would be better like this: "and install a customized module as a replacement"

**Review by xvik — CHANGES_REQUESTED** [review]
Thank you very much! Sorry for late reply!
Could you please correct a few points

**Comment by JKomoroski** [comment]
@xvik I've addressed your comments. Let me know if you have any other concerns.

**Comment by xvik** [comment]
Thank you!

**Pull request merged**